### PR TITLE
Let ProjectConnection extend Closeable interface

### PR DESCRIPTION
**Pull request by stefanleh**
ProjectConnection should extend Closeable interface for use of try-with-resources pattern.

Signed-off-by: Stefan <stefanleh@users.noreply.github.com>

### Contributor Checklist
- [x] [Review Contribution Guidelines](https://github.com/gradle/gradle/blob/master/CONTRIBUTING.md)
- [x] Make sure that all commits are [signed off](https://git-scm.com/docs/git-commit#git-commit---signoff) to indicate that you agree to the terms of [Developer Certificate of Origin](https://developercertificate.org/).
- [x] Check ["Allow edit from maintainers" option](https://help.github.com/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork/) in pull request so that additional changes can be pushed by Gradle team
- [x] Provide integration tests (under `<subproject>/src/integTest`) to verify changes from a user perspective
- [x] Provide unit tests (under `<subproject>/src/test`) to verify logic
- [x] Update User Guide, DSL Reference, and Javadoc for public-facing changes
- [x] Ensure that tests pass locally: `./gradlew <changed-subproject>:check`

### Gradle Core Team Checklist
- [x] Verify design and implementation 
- [x] Verify test coverage and CI build status
- [x] Verify documentation
- [ ] Recognize contributor in release notes

### CODE REVIEW

File: subprojects/tooling-api/src/main/java/org/gradle/tooling/ProjectConnection.java [hunk 1]
```diff
@@ -25,19 +26,15 @@
  *    .forProjectDirectory(new File("someFolder"))
  *    .connect();
  *
- * try {
- *    //obtain some information from the build
- *    BuildEnvironment environment = connection.model(BuildEnvironment.class)
- *      .get();
- *
+ * //obtain some information from the build
+ * try (BuildEnvironment environment = connection.model(BuildEnvironment.class).get()) {
```
Comment (oehme) [inline subprojects/tooling-api/src/main/java/org/gradle/tooling/ProjectConnection.java]: This doesn't make sense - The connection is closable, not the model.

**Comment by stefanleh** [comment]
Sorry, fixed it and rebased my mess into one commit again.

**Comment by stefanleh** [comment]
Any news here? I'd love to use try-with-resources with 4.9.

**Comment by oehme** [comment]
I kicked off a branch build and will merge for 4.9 once it succeeds.

**Comment by stefanleh** [comment]
Doesnt look good. What was the issue? Can i fix it?

**Comment by oehme** [comment]
The sample has a syntax error - it's missing a semicolon.

**Comment by stefanleh** [comment]
The sample gets compiled? Wow. I will fix it.

**Comment by oehme** [comment]
Yes, we make sure that all samples are working code, so users can copy-paste them into their projects.

**Comment by stefanleh** [comment]
Thats a real nice feature!
I've added the missing ; now.
Hope it works.

**Comment by oehme** [comment]
Thank you @stefanleh! I've merged this for 4.9 RC2

**Pull request merged**